jme3-core/src/main/java/com/jme3/renderer/RenderManager.java

```diff
@@ -1298,4 +1301,25 @@ public void render(float tpf, boolean mainFrameBufferActive) {
             }
         }
     }
+
+    /**
+     * Set a render filter. Every geometry will be tested against this filter
+     * before rendering and will only be rendered if the filter returns true.
+     * 
+     * @param filter
+     */
+    public void setRenderFilter(Function<Geometry, Boolean> filter) {

```

Comment:
In general I don't love temporary state like this that can (if mishandled) persist across operations (rendering env -> rendering main scene).

I find the following pattern for situations like this intuitive and easy to use correctly because it doesn't alter state:
```
render(<params>, filter) {
  ...
  if(filter != null && filter.test(n))
  ...
}

render(<params>){
  render(<params>, null);
}
```

Reply:
I generally agree with you, but the render manager was developed like some sort of state machine... goblin, there are too many render related methods to make passing a filter around efficient, also the same approach is used by many other methods (eg.   setForcedMaterial ).
I consider the RenderManager to be an internal api, so i wouldn't worry about this too much as long as code calling it does it correctly. Also, with this approach you can make interest things, like adapting existing code and libraries .